# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.cpp
# Commit: 7a28b63b4fc3
# Full Hash: 7a28b63b4fc3c64e95e4e1885909731e2c2df524
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-19 03:53:26
# Regressor Bug: 1839918
# File Overlap Count: 3
# Description:
#   Bug 1839918 - Pass BrowsingContext ID and third-party flag through CookieService. r=bvandersloot,cookie-reviewers,edgul
#   
#   Depends on D185302
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D185303
# ==============================================================================

diff -r 2bd8c91c5d97 -r 7a28b63b4fc3 netwerk/cookie/CookieService.cpp
--- a/netwerk/cookie/CookieService.cpp	Mon Sep 18 11:55:09 2023 +0000
+++ b/netwerk/cookie/CookieService.cpp	Mon Sep 18 11:55:09 2023 +0000
@@ -2481,7 +2481,9 @@
 bool CookieService::SetCookiesFromIPC(const nsACString& aBaseDomain,
                                       const OriginAttributes& aAttrs,
                                       nsIURI* aHostURI, bool aFromHttp,
-                                      const nsTArray<CookieStruct>& aCookies) {
+                                      const nsTArray<CookieStruct>& aCookies,
+                                      uint64_t aBrowsingContextId,
+                                      bool aIsThirdPartyCookie) {
   if (!IsInitialized()) {
     // If we are probably shutting down, we can ignore this cookie.
     return true;
@@ -2520,8 +2522,11 @@
     cookie->SetCreationTime(
         Cookie::GenerateUniqueCreationTime(currentTimeInUsec));
 
+    RefPtr<dom::BrowsingContext> browsingContext =
+        BrowsingContext::Get(aBrowsingContextId);
     storage->AddCookie(nullptr, aBaseDomain, aAttrs, cookie, currentTimeInUsec,
-                       aHostURI, ""_ns, aFromHttp, nullptr, false);
+                       aHostURI, ""_ns, aFromHttp, browsingContext,
+                       aIsThirdPartyCookie);
   }
 
   return true;